node/constructors.go

```diff
@@ -562,16 +559,16 @@ func NewSeedNode(config *cfg.Config,
 	return node, nil
 }
 
-// Option sets a parameter for the node.
-type Option func(*Node)
+// option sets a parameter for the node.
+type option func(*nodeImpl)
```

Comment:
I'm pretty sure we can get rid of these. It doesn't make any more sense to have these options if they are not exposed

Reply:
some of the tests use them, so it was eaiser to keep them around. We can definitely remove them later, particularly when we ditch the legacy p2p stuff.

Reply:
yeah but we have tests for them to make sure that they work for the users that use them. If there's no users using them then there's no reason for those tests either

Reply:
ok, you convinced me, removed.